src/main/java/decodes/gui/TimeSeriesChartFrame.java

```diff
@@ -0,0 +1,51 @@
+package decodes.gui;
+
+import decodes.tsdb.CTimeSeries;
+import decodes.tsdb.TimeSeriesIdentifier;
+import opendcs.dai.TimeSeriesDAI;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.*;
+import java.util.Calendar;
+
+public class TimeSeriesChartFrame extends JFrame
+{
+    private final static org.slf4j.Logger log = LoggerFactory.getLogger(TimeSeriesChartFrame.class);
+    private final TimeSeriesDAI timeSeriesDAO;
+    private final TimeSeriesIdentifier[] tsids;
+    public TimeSeriesChartFrame(TimeSeriesDAI timeSeriesDAO, TimeSeriesIdentifier[] tsids )
+    {
+        this.timeSeriesDAO = timeSeriesDAO;
+        this.tsids = tsids;
+    }
+
+    public void plot(){
```

Comment:
After looking at the below usage of plot, would `::plot(The data)` be better. Only one instance of the frame would need to be created and could then be reused.

Reply:
The next step in my mind was to add a panel with a time-window selection and a 'refresh button, to the TimeSeriesChartFrame, so the initial time window of 7 days could be changed by the user.

I think you may be suggesting another object TimeSeriesSeriesChartData that gets passed in, so the time-window selection would be elsewhere?

Reply:
No, you're passing into the constructor a DAO and a list. I'm suggestion that this function just takes "the data" already processed for rendering. 

You're refresh button would then be responsible for actually getting the data.

E.g. why does the ChartFrame need to know how to access the database? Seems like it could take only it's constructions, `TimeSeriesLine`.

Reply:
Because, when expanded later the Frame  (hopefully refactored later to have tsPlotController) will need to be able to update based on different time window inputs ( startDate:   ,  endDate) not yet implemented.    That logic isn't TsListPanel specific. 

The 'Refresh' button, start and end time inputs will also need a home.

Reply:
Okay, so your intention is to move that logic out later. Just when there's a more appropriate place?